data/block_export_data.go

```diff
@@ -29,6 +38,75 @@ type BlockData struct {
 	Certificate *agreement.Certificate
 }
 
+// MakeBlockDataFromValidatedBlock makes BlockData from agreement.ValidatedBlock
+func MakeBlockDataFromValidatedBlock(base BlockData, input ledgercore.ValidatedBlock) BlockData {
+	delta := input.Delta()
+	return BlockData{
+		BlockHeader: input.Block().BlockHeader,
+		Payset:      input.Block().Payset,
+		Delta:       &delta,
+		Certificate: base.Certificate,
+	}
+
+}
+
+// MakeBlockDataFromEncodedBlockCertificate makes BlockData from rpcs.EncodedBlockCert
+func MakeBlockDataFromEncodedBlockCertificate(base BlockData, input *rpcs.EncodedBlockCert) BlockData {
+	if input == nil {
+		return base
+	}
+	cert := input.Certificate
+
+	return BlockData{
+		BlockHeader: input.Block.BlockHeader,
+		Payset:      input.Block.Payset,
+		Delta:       base.Delta,
+		Certificate: &cert,
+	}
+}
+
+// MakeBlockDataFromBlock makes BlockData from boookeeping.Block
+func MakeBlockDataFromBlock(base BlockData, input bookkeeping.Block) BlockData {
+	return BlockData{
+		BlockHeader: input.BlockHeader,
+		Payset:      input.Payset,
+		Delta:       base.Delta,
+		Certificate: base.Certificate,
+	}
+}
```

Comment:
These seem mostly unnecessary, why not just create the struct normally?

Reply:
I think because it allows people to "merge" previous block data variables with new inputs easily.  I figured if we are building a pipeline that "adding" to the previous block data would be a common operation.

Reply:
That use case makes sense, how about putting them on the `BlockData` type?

i.e. one of these:
`func (bd *BlockData) UpdateBlock(block bookkeeping.Block)`
`func (bd BlockData) UpdateBlock(block bookkeeping.Block) BlockData`